stellargraph/data/explorer.py

```diff
@@ -796,6 +796,29 @@ class TemporalRandomWalk(GraphWalk):
     for the algorithm, only the relative differences (e.g. seconds, days, etc).
     """
 
+    def __init__(self, graph, graph_schema=None, seed=None):
+        super().__init__(graph, graph_schema=graph_schema, seed=seed)
+        self._edges, self._times = self.graph.edges(include_edge_weight=True)
+        self._validate_times()
```

Comment:
Instead of constructing the object and then validating, meaning it may be in an invalid state, what do you think about validating first?

```python
edges, times = self.graph.edges(include_edge_weight=True)

(neg_time_locs,) = np.where(times < 0)
if len(neg_time_locs) > 0:
    ...

# all good
self._edges = edges
self._times = times
```

Totally minor, though.

Reply:
Is your suggestion based more on readability or do you mean that if we assign `self._edges` and `self._times` first before running validation, it's somehow possible for the object to be instantiated in an invalid state?

Reply:
A bit of both, but I guess this class is simple enough that the invalid state probably isn't observable (I think the `self` value would have to have been saved somewhere externally accessible inside `__init__`, and then after the validation error is thrown, the saved instance accessed and used: I don't think this class does that saving anywhere, and thus we shouldn't worry).

Reply:
Hmm I see, I hadn't considered that! And didn't realise that would result in that externally accessible thing actually pointing to the semi-constructed object.. I've taken your suggestion just to be safe